### ISSUE TITLE
.Torrent downloaded files are bing corrupted by the Internet providers worldwide!!!

### ISSUE THREAD

**Issue by Enoch-X** [DPWSMedia/xbtit#33]
Hi everbody.
I've recently updated to XBTiT-2.5.3, thanks for all bugfixes!

Now we're facing a major issue! It has nothing to do with bugs in the script, but with a general problem for all Bittorrent Trackers scripts worldwide!

Recently, when someone tries to download a .torrent file we get a message: 
"Unable to load ...xyz.torrent" torrent is not valid bencoding!
We've tested many files, download over the Internet, via our Internet provider "Movistar" in Spain.

Everytime we get same message: 
"Unable to load ...xyz.torrent" torrent is not valid bencoding!

We've tested this with different Torrent clients like: uTorrent, Deluge and Transmission,.. they all state the same error message!

Now when we select a random .btf Torrent file from the XBTiT Torrent Tracker directory loacally (without downloading over the Internet via the Internet Service Provider "Movistar"!!!) and we change the .btf extension to .torrent and we load file into one of thye Torrent clients, everything works perfect without errors! 

So, our conclusion is that during the download process of a .torrent file, the ISP "Movistar is corrupting the .torrent files. And we think that this is a worldwide assault by the ISP companies!!! Their killing our Internet freedom man, we're under attack!

Now the workaround is the **Magnet Link**!

My request is if you, please, would like to add the magnet link to the XBTiT script, so that users have the option to download the .torrent file to their Torrent client or to use a Magnet Link as alternative?

I've searched the Internet and I found magnet link HTML script snippet from a PHPBB webpage:

`<script` type="text/javascript">

var str="{TEXT}";

if (str.match("dn="))
{
var fn=decodeURI("{TEXT}");
while (fn.indexOf("+")>0) fn=fn.replace("+", " ");

str1 = fn.split("&");
str2 = str1[1].slice(7);

var s="<img src='./images/magnet-icon.gif'> "+ " <a href='magnet:?xt=urn:btih:" + str +" '>" + str2 + "</a>  ";
document.write(s);
}

else
{
var s="<img src='./images/magnet-icon.gif'> "+" <a href='magnet:?xt=urn:btih:" + str +"'>Magnet Link</a>  ";
document.write(s);
}

</script>` 

Here is a link to the page containing the script [https://www.phpbb.com/community/viewtopic.php?f=71&t=2127808](url)

Thanks in advance for your coorporation!

You know: "Every problem has a solution, when one door is closed, aothers are open!"
God Bless!
Greetings from Tenerife, Spain ;-)

**Comment by Enoch-X** [comment]
Hi I took a good look at the content of the .torrent files in the torrent directory from new updated XBTit 2.5.3 script with Notepad ++
After the download the .torrent files contain a lot of code at the top, that is not supposed to be in the .torrent file.

The .torrent files all contain a single line brake `<br>` followed by a table row `<tr>` a total of 20 rows of XBTiT table data displayed in Notepd ++.
I didn't has this before. My question is if this could be a bug in the script or what?

**Comment by Enoch-X** [comment]
Hi, I found it!

The include/functions.php file of the new XBTiT 2.5.3 causes the problem!

I've replaced it with the include/functions.php file of the XBTiT 2.5.2 version and now everything workes fine. 

I don't know what causes the problem, the include/functions.php file of the XBTiT 2.5.2 version contains 1854 lines of code in Notepad ++ and the include/functions.php from the latest version 2.5.3 contains 1860 lines of code in Notepad++ so it's not easy to find the part that causes the issue.

But i've replaced the new functions.php with the older version of functions.php and the download of the torrent files don't cause any more troubles.

Problem solved!
